### ISSUE TITLE
Is MQBoost suitable for health insurance medical loss prediction? Tips for heavy-tailed claim data?

### ISSUE THREAD

**Issue by vvvviii** [RektPunk/MQBoost#52]
Hi @RektPunk,

I'm using MQBoost for health insurance medical loss prediction. Is this library suitable for this task? Also, could you share some tips for applying it to heavy-tailed medical claim data?

Thank you for your help!

**Comment by RektPunk** [comment]
Hi @vvvviii, 

Thank you for your interest in MQBoost! I believe it is suitable for medical loss prediction.

While my primary background is in data science and financial analysis, MQBoost’s core mechanism—using an objective function with constraints to prevent quantile crossing—is a universal solution for maintaining logical consistency in uncertainty estimation. Even for volatile medical claims, the predicted quantiles will remain perfectly ordered. **Careful hyper parameter tuning** is essential though, especially for heavy-tailed data where finding the right balance between bias and variance is key.

Also, I’m excited to share that the formal paper detailing the MQBoost methodology is published in the ***Machine Learning*** journal: [*Monotone Composite Quantile Regression via Second-Order Gradient Boosting Framework*](https://doi.org/10.1007/s10994-026-07058-2). It will provide more in-depth technical details on the constrained optimization approach.  Hope this helps! Let me know if you have any further questions.

Here is a minimal example tailored to your use case:

<img width="1474" height="892" alt="Image" src="https://github.com/user-attachments/assets/62240271-f41c-44f7-9286-d812cac5132f" />
<details>
<summary><b>Click to expand the example code</b></summary>

```python
import matplotlib.pyplot as plt
import numpy as np
from mqboost import MQDataset, MQRegressor

np.random.seed(42)
n_samples = 100
x = np.linspace(0, 10, n_samples)
y = np.exp(np.random.normal(0, 1, n_samples))

alphas = [0.01, 0.05, 0.1, 0.5, 0.9, 0.95, 0.99]
lgb_params = {
    "max_depth": 10,
    "num_leaves": 50,
    "learning_rate": 0.3,
    "boosting_type": "gbdt",
}
mq_regressor = MQRegressor(params=lgb_params, objective="approx", model="lightgbm")
train_dataset = MQDataset(
    data=x.reshape(-1, 1), label=y, alphas=alphas, model="lightgbm"
)
mq_regressor.fit(dataset=train_dataset)

preds = mq_regressor.predict(train_dataset)

plt.figure(figsize=(10, 6))
plt.scatter(x, y, color="black", s=15, alpha=0.5, label="Actual Claims")
colors = plt.cm.Blues(np.linspace(0.4, 1, len(alphas)))
for i, alpha in enumerate(alphas):
    plt.plot(
        x,
        preds[i, :],
        color=colors[i],
        label=f"q_{alpha}",
        linewidth=2 if alpha == 0.5 else 1,
    )
plt.fill_between(x, preds[0, :], preds[-1, :], color="blue", alpha=0.1)
plt.title("MQBoost: Medical Loss Prediction")
plt.legend(loc="upper left", bbox_to_anchor=(1, 1))
plt.grid(True, which="both", alpha=0.3)
plt.tight_layout()
plt.show()
```

**Comment by RektPunk** [comment]
Closing this issue as the question seems to be resolved. Best of luck with your research @vvvviii,  and don't hesitate to reach out if you need further assistance!
